server/events.go

```diff
@@ -767,6 +806,25 @@ func (s *Server) statszReq(sub *subscription, _ *client, subject, reply string,
 	s.sendStatsz(reply)
 }
 
+func (s *Server) zReq(reply string, msg []byte, optz interface{}, respf func() (interface{}, error)) {
+	if !s.EventsEnabled() || reply == _EMPTY_ {
+		return
+	}
+	server := &ServerInfo{}
+	response := map[string]interface{}{"server": server}
+	var err error
+	if len(msg) != 0 {
+		err = json.Unmarshal(msg, optz)
+	}
+	if err == nil {
+		response["data"], err = respf()
+	}
+	if err != nil {
+		response["error"] = err.Error()
+	}
+	s.sendInternalMsgLocked(reply, _EMPTY_, server, &response)
```

Comment:
Not sure you need pointer to `response` here.

Reply:
removed the pointer